ivadomed/main.py

```diff
@@ -1,649 +1,214 @@
-        transformation_dict = context["transformation_validation"]
+def display_selected_transfoms(params, dataset_type):
+    """Display in terminal the selected transforms for a given dataset.
 
-    # Compose Testing transforms
-    val_transform = imed_transforms.Compose(transformation_dict, requires_undo=True)
+    Args:
+        params (dict):
+        dataset_list (list): e.g. ['testing'] or ['training', 'validation']
+    Returns:
+        None
+    """
+    print('\nSelected transformations for the {} dataset:'.format(dataset_type))
+    for k in list(params.keys()):
+        print('\t{}: {}'.format(k, params[k]))
 
-    # inverse transformations
-    val_undo_transform = imed_transforms.UndoCompose(val_transform)
 
-    if context.get("split_path") is None:
-        test_lst = joblib.load("./" + context["log_directory"] + "/split_datasets.joblib")['test']
-    else:
-        test_lst = joblib.load(context["split_path"])['test']
+def get_subdatasets_transforms(transform_params):
+    """Get transformation parameters for each subdataset: training, validation and testing.
 
-    ds_test = imed_loader.load_dataset(test_lst, val_transform, context)
-
-    # if ROICrop2D in transform, then apply SliceFilter to ROI slices
-    if 'ROICrop2D' in context["transformation_validation"].keys():
-        ds_test = imed_loader_utils.filter_roi(ds_test, nb_nonzero_thr=context["slice_filter_roi"])
+    Args:
+        transform_params (dict):
+    Returns:
+        dict, dict, dict
+    """
+    train, valid, test = {}, {}, {}
+    subdataset_default = ["training", "validation", "testing"]
+    # Loop across transformations
+    for transform_name in transform_params:
+        subdataset_list = ["training", "validation", "testing"]
+        # Only consider subdatasets listed in dataset_type
+        if "dataset_type" in transform_params[transform_name]:
+            subdataset_list = transform_params[transform_name]["dataset_type"]
+        # Add current transformation to the relevant subdataset transformation dictionaries
+        for subds_name, subds_dict in zip(subdataset_default, [train, valid, test]):
+            if subds_name in subdataset_list:
+                subds_dict[transform_name] = transform_params[transform_name]
+                if "dataset_type" in subds_dict[transform_name]:
+                    del subds_dict[transform_name]["dataset_type"]
+    return train, valid, test
 
-    if film_bool:  # normalize metadata before sending to network
-        metadata_clustering_models = joblib.load(
-            "./" + context["log_directory"] + "/clustering_models.joblib")
 
-        ds_test = imed_film.normalize_metadata(ds_test,
-                                               metadata_clustering_models,
-                                               context["debugging"],
-                                               context["metadata"],
-                                               False)
+def run_main():
+    if len(sys.argv) <= 1:
```

Comment:
This line checks if there is one parameter missing, we should print an error message here, no? + The print don't print the config file only a string. Do you agree with this change: if len(sys.argv) != 2: print(error message)

Reply:
Amen!
Done:
```python
    if len(sys.argv) != 2:
        print("\nERROR: Please indicate the path of your configuration file, "
              "e.g. ivadomed ivadomed/config/config.json\n")
        return
    path_config_file = sys.argv[1]
    if not os.path.isfile(path_config_file) and not path_config_file.endswith('.json'):
        print("\nERROR: The provided configuration file path (.json) is invalid: {}\n".format(path_config_file))
        return
```